<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1600</width>
    <height>900</height>
   </rect>
  </property>
  <property name="font">
   <font>
    <family>HarmonyOS Sans SC</family>
    <pointsize>12</pointsize>
    <stylestrategy>PreferAntialias</stylestrategy>
   </font>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QLabel" name="label">
      <property name="text">
       <string>寄存器:</string>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0">
       <widget class="QTableWidget" name="tableWidget_RegistersAreas">
        <property name="font">
         <font>
          <family>Cascadia Code</family>
          <pointsize>12</pointsize>
          <stylestrategy>PreferAntialias</stylestrategy>
         </font>
        </property>
        <property name="verticalScrollBarPolicy">
         <enum>Qt::ScrollBarPolicy::ScrollBarAlwaysOn</enum>
        </property>
        <property name="horizontalScrollBarPolicy">
         <enum>Qt::ScrollBarPolicy::ScrollBarAlwaysOff</enum>
        </property>
        <property name="editTriggers">
         <set>QAbstractItemView::EditTrigger::NoEditTriggers</set>
        </property>
        <property name="alternatingRowColors">
         <bool>true</bool>
        </property>
        <property name="selectionMode">
         <enum>QAbstractItemView::SelectionMode::SingleSelection</enum>
        </property>
        <property name="selectionBehavior">
         <enum>QAbstractItemView::SelectionBehavior::SelectRows</enum>
        </property>
        <property name="gridStyle">
         <enum>Qt::PenStyle::NoPen</enum>
        </property>
        <property name="wordWrap">
         <bool>false</bool>
        </property>
        <column>
         <property name="text">
          <string>FIRST ADDRESS</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>BYTES</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>TYPE</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>WRITE TIMES</string>
         </property>
        </column>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QTableWidget" name="tableWidget_Registers">
        <property name="font">
         <font>
          <family>Cascadia Code</family>
          <pointsize>12</pointsize>
          <stylestrategy>PreferAntialias</stylestrategy>
         </font>
        </property>
        <property name="verticalScrollBarPolicy">
         <enum>Qt::ScrollBarPolicy::ScrollBarAlwaysOn</enum>
        </property>
        <property name="horizontalScrollBarPolicy">
         <enum>Qt::ScrollBarPolicy::ScrollBarAlwaysOff</enum>
        </property>
        <property name="editTriggers">
         <set>QAbstractItemView::EditTrigger::NoEditTriggers</set>
        </property>
        <property name="alternatingRowColors">
         <bool>true</bool>
        </property>
        <property name="selectionMode">
         <enum>QAbstractItemView::SelectionMode::SingleSelection</enum>
        </property>
        <property name="selectionBehavior">
         <enum>QAbstractItemView::SelectionBehavior::SelectRows</enum>
        </property>
        <property name="gridStyle">
         <enum>Qt::PenStyle::NoPen</enum>
        </property>
        <property name="wordWrap">
         <bool>false</bool>
        </property>
        <attribute name="horizontalHeaderStretchLastSection">
         <bool>true</bool>
        </attribute>
        <column>
         <property name="text">
          <string>INDEX</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>ADDRESS</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>R/W</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>DATA</string>
         </property>
        </column>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QLabel" name="label_2">
      <property name="text">
       <string>输入数据</string>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="3" column="5">
       <widget class="QPushButton" name="pushButton_Prase">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>解析</string>
        </property>
       </widget>
      </item>
      <item row="1" column="4">
       <widget class="QLineEdit" name="lineEdit_Data">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>28</height>
         </size>
        </property>
        <property name="font">
         <font>
          <family>Cascadia Mono</family>
          <pointsize>12</pointsize>
          <stylestrategy>PreferAntialias</stylestrategy>
         </font>
        </property>
        <property name="text">
         <string>01 FF FF</string>
        </property>
       </widget>
      </item>
      <item row="0" column="3">
       <widget class="QLabel" name="label_9">
        <property name="font">
         <font>
          <family>HarmonyOS Sans SC</family>
          <pointsize>12</pointsize>
          <stylestrategy>PreferAntialias</stylestrategy>
          <fontweight>Black</fontweight>
         </font>
        </property>
        <property name="text">
         <string>DATA / AMOUNT</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignmentFlag::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLabel" name="label_5">
        <property name="font">
         <font>
          <family>HarmonyOS Sans SC</family>
          <pointsize>12</pointsize>
          <stylestrategy>PreferAntialias</stylestrategy>
          <fontweight>Black</fontweight>
         </font>
        </property>
        <property name="text">
         <string>FUNCTION</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignmentFlag::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QSpinBox" name="spinBox_DeviceAddress">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="font">
         <font>
          <family>Cascadia Mono</family>
          <pointsize>12</pointsize>
          <stylestrategy>PreferAntialias</stylestrategy>
         </font>
        </property>
        <property name="alignment">
         <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
        </property>
        <property name="buttonSymbols">
         <enum>QAbstractSpinBox::ButtonSymbols::NoButtons</enum>
        </property>
        <property name="suffix">
         <string/>
        </property>
        <property name="maximum">
         <number>255</number>
        </property>
        <property name="value">
         <number>1</number>
        </property>
        <property name="displayIntegerBase">
         <number>16</number>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QSpinBox" name="spinBox_RegisterAddress">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="font">
         <font>
          <family>Cascadia Mono</family>
          <pointsize>12</pointsize>
          <stylestrategy>PreferAntialias</stylestrategy>
         </font>
        </property>
        <property name="alignment">
         <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
        </property>
        <property name="buttonSymbols">
         <enum>QAbstractSpinBox::ButtonSymbols::NoButtons</enum>
        </property>
        <property name="suffix">
         <string/>
        </property>
        <property name="maximum">
         <number>65535</number>
        </property>
        <property name="value">
         <number>100</number>
        </property>
        <property name="displayIntegerBase">
         <number>16</number>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QComboBox" name="comboBox_FunctionCode">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="font">
         <font>
          <family>Cascadia Mono</family>
          <pointsize>12</pointsize>
          <stylestrategy>PreferAntialias</stylestrategy>
         </font>
        </property>
        <property name="editable">
         <bool>true</bool>
        </property>
        <property name="currentIndex">
         <number>6</number>
        </property>
        <item>
         <property name="text">
          <string>01 Read Multiple Coils</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>02 Read Multiple Discrete Inputs</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>03 Read Multiple Holding Registers</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>04 Read Multiple Input Registers</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>05 Write Single Coil</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>06 Write Single Holding Registers</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>0F Write Multiple Coils</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>10 Write Multiple Holding Registers</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="3" column="0" colspan="5">
       <widget class="QLineEdit" name="lineEdit_Frame">
        <property name="font">
         <font>
          <family>Cascadia Mono</family>
          <pointsize>12</pointsize>
          <stylestrategy>PreferAntialias</stylestrategy>
         </font>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QLabel" name="label_6">
        <property name="font">
         <font>
          <family>HarmonyOS Sans SC</family>
          <pointsize>12</pointsize>
          <stylestrategy>PreferAntialias</stylestrategy>
          <fontweight>Black</fontweight>
         </font>
        </property>
        <property name="text">
         <string>REGISTER ADDRESS</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignmentFlag::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="label_4">
        <property name="font">
         <font>
          <family>HarmonyOS Sans SC</family>
          <pointsize>12</pointsize>
          <stylestrategy>PreferAntialias</stylestrategy>
          <fontweight>Black</fontweight>
         </font>
        </property>
        <property name="text">
         <string>DEVICE ADDRESS</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignmentFlag::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="1" column="5">
       <widget class="QLineEdit" name="lineEdit_CRC">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Ignored" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="font">
         <font>
          <family>Cascadia Mono</family>
          <pointsize>12</pointsize>
          <stylestrategy>PreferAntialias</stylestrategy>
         </font>
        </property>
        <property name="maxLength">
         <number>5</number>
        </property>
        <property name="alignment">
         <set>Qt::AlignmentFlag::AlignCenter</set>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="3">
       <widget class="QSpinBox" name="spinBox_Amount">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="font">
         <font>
          <family>Cascadia Mono</family>
          <pointsize>12</pointsize>
          <stylestrategy>PreferAntialias</stylestrategy>
         </font>
        </property>
        <property name="alignment">
         <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
        </property>
        <property name="buttonSymbols">
         <enum>QAbstractSpinBox::ButtonSymbols::NoButtons</enum>
        </property>
        <property name="accelerated">
         <bool>false</bool>
        </property>
        <property name="maximum">
         <number>65535</number>
        </property>
        <property name="value">
         <number>8</number>
        </property>
        <property name="displayIntegerBase">
         <number>16</number>
        </property>
       </widget>
      </item>
      <item row="0" column="4">
       <widget class="QLabel" name="label_8">
        <property name="font">
         <font>
          <family>HarmonyOS Sans SC</family>
          <pointsize>12</pointsize>
          <stylestrategy>PreferAntialias</stylestrategy>
          <fontweight>Black</fontweight>
         </font>
        </property>
        <property name="text">
         <string>DATA</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignmentFlag::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="0" column="5">
       <widget class="QLabel" name="label_7">
        <property name="font">
         <font>
          <family>HarmonyOS Sans SC</family>
          <pointsize>12</pointsize>
          <stylestrategy>PreferAntialias</stylestrategy>
          <fontweight>Black</fontweight>
         </font>
        </property>
        <property name="text">
         <string>CRC</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignmentFlag::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="2" column="0" colspan="6">
       <widget class="QLabel" name="label_10">
        <property name="text">
         <string>帧</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QLabel" name="label_3">
      <property name="text">
       <string>解析结果</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QTableWidget" name="tableWidget_Result">
      <property name="font">
       <font>
        <family>Cascadia Code</family>
        <pointsize>12</pointsize>
        <stylestrategy>PreferAntialias</stylestrategy>
       </font>
      </property>
      <property name="editTriggers">
       <set>QAbstractItemView::EditTrigger::NoEditTriggers</set>
      </property>
      <property name="alternatingRowColors">
       <bool>true</bool>
      </property>
      <property name="selectionMode">
       <enum>QAbstractItemView::SelectionMode::SingleSelection</enum>
      </property>
      <property name="selectionBehavior">
       <enum>QAbstractItemView::SelectionBehavior::SelectRows</enum>
      </property>
      <property name="gridStyle">
       <enum>Qt::PenStyle::NoPen</enum>
      </property>
      <property name="wordWrap">
       <bool>false</bool>
      </property>
      <attribute name="horizontalHeaderStretchLastSection">
       <bool>true</bool>
      </attribute>
      <column>
       <property name="text">
        <string>Key</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Value</string>
       </property>
      </column>
     </widget>
    </item>
    <item>
     <widget class="QLabel" name="label_11">
      <property name="text">
       <string>返回帧</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QLineEdit" name="lineEdit_ReturnFrame">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>28</height>
       </size>
      </property>
      <property name="font">
       <font>
        <family>Cascadia Mono</family>
        <pointsize>12</pointsize>
        <stylestrategy>PreferAntialias</stylestrategy>
       </font>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
